src/client.spec.ts

```diff
@@ -842,14 +861,54 @@ describe('client', () => {
   });
 
   describe('getProvider', () => {
-    it('should get one provider', async () => {
-      const mockResult: ProviderResponse = {
+    it('should get provider in flat format', async () => {
+      const mockResult = {
         url: 'testUrl',
         name: 'testName',
         services: [{ id: 'default', baseUrl: 'http://superface.test/api' }],
         defaultService: 'default',
       };
 
+      const mockResponse = {
+        ok: true,
+        json: async () => mockResult,
+      };
+      const fetchMock = jest
+        .spyOn(client, 'fetch')
+        .mockResolvedValue(mockResponse as Response);
+      await expect(client.getProvider('test')).resolves.toEqual({
+        provider_id: 'testName',
+        url: 'testUrl',
+        definition: {
+          name: 'testName',
+          services: [{ id: 'default', baseUrl: 'http://superface.test/api' }],
+          defaultService: 'default',
+        },
+      });
+      expect(fetchMock).toBeCalledTimes(1);
+      expect(fetchMock).toBeCalledWith('/providers/test', {
+        authenticate: false,
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    });
+
+    it('should get provider in nested format', async () => {
+      const mockResult = {
+        url: 'testUrl',
```

Comment:
Minor thing, but I think provider id is missing here

```suggestion
        provider_id: 'testName',
        url: 'testUrl',
```